### DISCUSSION TITLE
Should we update Grafana tutorial to use this CLI?

### DISCUSSION THREAD

**Discussion by hongbo-miao** [grafana/plugin-tools#131] [Q&A] [answered]
I originally asked [What is the recommended way to create Grafana plugin?](https://community.grafana.com/t/what-is-the-recommended-way-to-create-grafana-plugin/75315).

Took me a while to finally found this repo: `@grafana/toolkit` -> [grafana-plugin-examples](https://github.com/grafana/grafana-plugin-examples) -> `grafana/create-plugin` 😅

Maybe worth replacing existing [tutorial](https://grafana.com/docs/grafana/latest/developers/plugins/) with this new CLI? 😃

**Comment by hongbo-miao** [comment]
I created an issue on grafana/grafana repo at https://github.com/grafana/grafana/issues/58191
Hopefully get "fixed". 😃

**Comment by jackw** [comment, marked as answer]
Hello @Hongbo-Miao, thanks for creating this discussion. We have updated the doc commands to use `create-plugin` and `sign-plugin`. They can currently be viewed under the `next` release channel: https://grafana.com/docs/grafana/next/developers/plugins/
Latest should update when Grafana `9.3.0` is released (which should be soon).

**Reply by danads-labs** [reply]
Is it me or is this tutorial outdated? I cannot find this code:
`async query(options: DataQueryRequest<MyQuery>): Promise<DataQueryResponse> {
  const { range } = options;
  const from = range!.from.valueOf();
  const to = range!.to.valueOf();

  const data = options.targets.map(target => {
    // Your code goes here.
  });

  return { data };
}`
after scaffolding a new data-source plugin using: https://grafana.com/tutorials/build-a-data-source-plugin/

**Comment by krait8064** [comment]
@jackw Can plugin tool be used for any grafana version,such as both 6.6.0 and 9.3.0 compatible for plugin tools. I can't find docs for compatible in github and grafana docs
